scripts/tools/memory/platform/linux.cfg

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Memory tools default configuation for Linux.
+
+{
+    'section': {
+        # By default, only these sections will be included
+        # when operating by sections.
+        'default': [
+            '.text', '.data', '.data.rel.ro', '.bss', '.dynamic', '.got',
+            '.init', '.init_array', '.rodata'
+        ]
+    },
+#   'symbol': {
+#       'free': {
+#           # These symbols mark the start or end of areas where memory that
+#           # does not belong to any symbol is considered unused (rather than
+#           # a gap that may be in use for some non-symbol purpose, e.g. string
+#           # constants or alignment).
+#           'start': [],
+#           'end': [],
+#       }
+#   },
+    'region': {
+        # Regions are sets of sections that can be used for aggregate reports.
+        'sections': {
+            'FLASH': [
+                ".dynstr",
+                ".dynsym",
+                ".eh_frame_hdr",
+                ".eh_frame",
+                ".fini",
+                ".gcc_except_table",
+                ".gnu.version_d",
+                ".gnu.version_r",
+                ".gnu.version",
+                ".hash",
+                ".init",
+                ".interp",
+                ".note.ABI-tag",
+                ".rodata1",
+                ".rodata",
+                ".strtab",
+                ".symtab",
+                ".text",
+            ],
+            'RAM': [
+                ".bss",
+                ".ctors",
+                ".data1",
+                ".data.rel.ro",
+                ".data",
```

Comment:
This also goes into flash right? Should it be listed there as well?

Reply:
Honestly shouldn't we use the program headers to determine the file size and ram size?

You can't tell from the name...

Reply:
> This also goes into flash right? Should it be listed there as well?

It's marked SHF_WRITE, so I put it under ‘RAM’.

> Honestly shouldn't we use the program headers to determine the file size and ram size?

Yes. This is actually a slightly different facility, for arbitrarily grouping sections (from a previous round of memory investigations, not new for this PR.) An actual segment report would be good, but I haven't gotten to that yet.

Reply:
I think it would make sense for the reports to include both the aggregation of loaded segments (generally as read-only vs read/write) and platform-specific ‘important’ sections (e.g. .text/.data/.bss or equivalents).